### PR TITLE
Javascript javascript1 week1

### PR DESCRIPTION
**Pull request by Rajesh-Bhatt**
Implemented the following tasks as JavaScript week1 Homework:

1. Man Future Age Calculator
2. Dog Age Calculator
3. House Price Estimator
4. Startup Names Generator

### CODE REVIEW

File: javascript/javascript1/week1/dogAge.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//2. Goodboy-Oldboy (A dog age calculator)
+
+let dogYearOfBirth = 2016;
+let dogYearFuture = 2027;
+let shouldShowResultInDogYears = true;
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/dogAge.js]: All those could be set as `const` as you are not reassigning them later. Remember to always use `const` until it doesn't `let` you to ;)

File: javascript/javascript1/week1/dogAge.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+let dogYear;
+if (shouldShowResultInDogYears) {
+  dogYear = dogYearFuture - dogYearOfBirth;
+} else {
+  dogYear = (dogYearFuture - dogYearOfBirth) * 7;
+}
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/dogAge.js]: 1. It's a good practice to always set a value to a variable, even when it's initiated as empty. For example you know that `dogYear` will be a number. So here you could set it as `let dogYear = 0;`
2. But on another end you repeat `dogYearFuture - dogYearOfBirth` two times. Why not set already on line 7 `dogYear` to `dogYearFuture - dogYearOfBirth`? Then you can also remove this `if else` and use `dogYear = dogYear * 7;` right before the `console.log` for the true case in the `if else` starting at line 14.

File: javascript/javascript1/week1/housePrice.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+//3. Housey pricey (A house price estimator)
+
+// Peter's house parameters
+let peterHouseWidth = 8;
+let peterHouseHeight = 10;
+let peterHouseDepth = 10;
+let peterHouseGardenSize = 100;
+let peterHouseRealPrice = 2500000;
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/housePrice.js]: The variable names are long and descriptive, which is a very good practice. However, you can define them as `const` as they won't change.

File: javascript/javascript1/week1/housePrice.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+// Julia's house parameters
+let juliaHouseWidth = 5;
+let juliaHouseHeigth = 8;
+let juliaHouseDepth = 11;
+let juliaHouseGardenSize = 70;
+let juliaHouseRealPrice = 1000000;
+
+let peterVolume = peterHouseWidth * peterHouseHeight * peterHouseDepth
+let juliaVolume = juliaHouseWidth * juliaHouseHeigth * juliaHouseDepth
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/housePrice.js]: Would be more correct to call those `peterHouseVolume` and `juliaHouseVolume`.

File: javascript/javascript1/week1/manAge.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+//1. Age-ify (A future age calculator)
+
+let yearOfBirth = 1983;
+let futureYear = 2027;
+
+let age = futureYear - yearOfBirth;
+
+//console.log("You will be " +age+ " years old in " +futureYear);
+console.log(`You will be ${age} years old in ${futureYear}.`);
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/manAge.js]: Perfect choice - [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are so much cleaner than string concatenations.

File: javascript/javascript1/week1/startupNames.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+//4. Ez Namey (Startup name generator) Optional
+
+
+//First array Names:
+let firstWords = [
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/startupNames.js]: Those arrays can also be set as `const` as you are not changing them.

File: javascript/javascript1/week1/startupNames.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    "Concepts",
+    "Technologies",
+    "Systems",
+    "Preparation",
+    "Inventions",
+    "Process",
+    "Tools",
+    "Product",
+  ];
+  const randomNumber = Math.floor(Math.random() * 10);
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/startupNames.js:31]: Using a single random number for both first and second words would give you always the same pairs. If you add another variable for a second random number that you can use at `secondWords`, you would get much bigger variety of combinations.

**Review by jason-vasilev — APPROVED** [review]
The code works as it's supposed to. Yet, there are a few things you can improve on.
- Mostly, remember to use `const` instead of `let`. Use `let` only when you start getting errors for not being able to reassign a variable.
- Some variable names can be made more descriptive and clear
- There is a small optimization you can do for the Dog Age task

You are welcome to merge now, as none of those suggestions are critical. However, it could be a good exercise to go through some of them for the practice.

File: javascript/javascript1/week1/dogAge.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+//2. Goodboy-Oldboy (A dog age calculator)
+
+const dogYearOfBirth = 2016;
+const dogYearFuture = 2027;
+const shouldShowResultInDogYears = true;
+
+const dogYear;
```
Comment (jason-vasilev) [inline javascript/javascript1/week1/dogAge.js]: This wouldn't work, as you are trying to reassign it. You would have gotten an error if you tried to run your code.

File: javascript/javascript1/week1/dogAge.js (hunk 1 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/dogAge.js]: I have changed all variable...

File: javascript/javascript1/week1/dogAge.js (hunk 2 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/dogAge.js]: I have optimized code here. Updated!!

File: javascript/javascript1/week1/housePrice.js (hunk 3 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/housePrice.js]: Changed from let to const...

File: javascript/javascript1/week1/housePrice.js (hunk 4 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/housePrice.js]: changed to peterHouseVolume  and juliaHouseVolume...

File: javascript/javascript1/week1/manAge.js (hunk 5 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/manAge.js]: Thank you..

File: javascript/javascript1/week1/startupNames.js (hunk 6 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/startupNames.js]: Done..

File: javascript/javascript1/week1/startupNames.js (hunk 7 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/startupNames.js:31]: I do not understand above comment, how to do that?

File: javascript/javascript1/week1/dogAge.js (hunk 8 above)
Comment (Rajesh-Bhatt) [reply javascript/javascript1/week1/dogAge.js]: changed to let from const in line no. 7. 
Thank you!!

**Pull request merged**